Error building from AUR

I reported the error on AUR as well but I did not get eny response so I'll post here as well:
```

Synchronizing package databases...
Starting full system upgrade...
Cloning onionshare build files...
Checking onionshare dependencies...
Building onionshare...
==> Making package: onionshare 2.0-2 (dom 24 feb 2019 12:24:04 CET)
==> Checking runtime dependencies...
==> Checking buildtime dependencies...
==> Retrieving sources...
  -> Downloading onionshare-2.0.tar.gz...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100   126    0   126    0     0    116      0 --:--:--  0:00:01 --:--:--   116

  0     0    0     0    0     0      0      0 --:--:--  0:00:01 --:--:--     0
100  807k    0  807k    0     0   284k      0 --:--:--  0:00:02 --:--:--  755k
100 2091k    0 2091k    0     0   665k      0 --:--:--  0:00:03 --:--:-- 1527k
==> Validating source files with sha512sums...
    onionshare-2.0.tar.gz ... Passed
==> Removing existing $srcdir/ directory...
==> Extracting sources...
  -> Extracting onionshare-2.0.tar.gz with bsdtar
==> Starting check()...
============================= test session starts ==============================
platform linux -- Python 3.7.2, pytest-4.3.0, py-1.7.0, pluggy-0.8.1
rootdir: /tmp/pamac-build/onionshare/src/onionshare-2.0, inifile:
collected 281 items

tests/local_onionshare_404_public_mode_skips_ratelimit_test.py s         [  0%]
tests/local_onionshare_404_triggers_ratelimit_test.py s                  [  0%]
tests/local_onionshare_quitting_during_share_prompts_warning_test.py s   [  1%]
tests/local_onionshare_receive_mode_sender_closed_test.py s              [  1%]
tests/local_onionshare_receive_mode_timer_test.py s                      [  1%]
tests/local_onionshare_receive_mode_upload_non_writable_dir_test.py s    [  2%]
tests/local_onionshare_receive_mode_upload_public_mode_non_writable_dir_test.py s [  2%]
                                                                         [  2%]
tests/local_onionshare_receive_mode_upload_public_mode_test.py s         [  2%]
tests/local_onionshare_receive_mode_upload_test.py s                     [  3%]
tests/local_onionshare_settings_dialog_legacy_tor_test.py s              [  3%]
tests/local_onionshare_settings_dialog_no_tor_test.py s                  [  3%]
tests/local_onionshare_settings_dialog_v3_tor_test.py s                  [  4%]
tests/local_onionshare_share_mode_download_public_mode_test.py s         [  4%]
tests/local_onionshare_share_mode_download_stay_open_test.py s           [  4%]
tests/local_onionshare_share_mode_download_test.py s                     [  5%]
tests/local_onionshare_share_mode_large_download_test.py s               [  5%]
tests/local_onionshare_share_mode_slug_persistent_test.py s              [  6%]
tests/local_onionshare_share_mode_timer_test.py s                        [  6%]
tests/local_onionshare_share_mode_timer_too_short_test.py s              [  6%]
tests/local_onionshare_share_mode_unreadable_file_test.py s              [  7%]
tests/onionshare_790_cancel_on_second_share_test.py s                    [  7%]
tests/onionshare_receive_mode_upload_public_mode_test.py s               [  7%]
tests/onionshare_receive_mode_upload_test.py s                           [  8%]
tests/onionshare_share_mode_cancel_share_test.py s                       [  8%]
tests/onionshare_share_mode_download_public_mode_test.py s               [  8%]
tests/onionshare_share_mode_download_stay_open_test.py s                 [  9%]
tests/onionshare_share_mode_download_test.py s                           [  9%]
tests/onionshare_share_mode_persistent_test.py s                         [  9%]
tests/onionshare_share_mode_stealth_test.py s                            [ 10%]
tests/onionshare_share_mode_timer_test.py s                              [ 10%]
tests/onionshare_share_mode_tor_connection_killed_test.py s              [ 11%]
tests/onionshare_share_mode_v2_onion_test.py s                           [ 11%]
tests/test_onionshare.py .......                                         [ 13%]
tests/test_onionshare_common.py ........................................ [ 28%]
.......................................................................  [ 53%]
tests/test_onionshare_settings.py ............                           [ 57%]
tests/test_onionshare_strings.py ....                                    [ 59%]
tests/test_onionshare_web.py FFFF....................................... [ 74%]
........................................................................ [100%]

=================================== FAILURES ===================================
___________________________ TestWeb.test_share_mode ____________________________

self = <tests.test_onionshare_web.TestWeb object at 0x7f6a18dcc2b0>
common_obj = <onionshare.common.Common object at 0x7f6a18dcc7b8>

    def test_share_mode(self, common_obj):
        web = web_obj(common_obj, 'share', 3)
        assert web.mode is 'share'
        with web.app.test_client() as c:
            # Load 404 pages
            res = c.get('/')
            res.get_data()
>           assert res.status_code == 404
E           assert 200 == 404
E            +  where 200 = <Response 1634 bytes [200 OK]>.status_code

tests/test_onionshare_web.py:77: AssertionError
____________ TestWeb.test_share_mode_close_after_first_download_on _____________

self = <tests.test_onionshare_web.TestWeb object at 0x7f6a18ebbeb8>
common_obj = <onionshare.common.Common object at 0x7f6a1e102208>
temp_file_1024 = '/tmp/tmp1v1nzuzb'

    def test_share_mode_close_after_first_download_on(self, common_obj, temp_file_1024):
        web = web_obj(common_obj, 'share', 3)
        web.stay_open = False
    
        assert web.running == True
    
        with web.app.test_client() as c:
            # Download the first time
            res = c.get('/{}/download'.format(web.slug))
            res.get_data()
>           assert res.status_code == 200
E           assert 404 == 200
E            +  where 404 = <Response 501 bytes [404 NOT FOUND]>.status_code

tests/test_onionshare_web.py:104: AssertionError
____________ TestWeb.test_share_mode_close_after_first_download_off ____________

self = <tests.test_onionshare_web.TestWeb object at 0x7f6a18ef80b8>
common_obj = <onionshare.common.Common object at 0x7f6a18ef8160>
temp_file_1024 = '/tmp/tmpl7w2g9wo'

    def test_share_mode_close_after_first_download_off(self, common_obj, temp_file_1024):
        web = web_obj(common_obj, 'share', 3)
        web.stay_open = True
    
        assert web.running == True
    
        with web.app.test_client() as c:
            # Download the first time
            res = c.get('/{}/download'.format(web.slug))
            res.get_data()
>           assert res.status_code == 200
E           assert 404 == 200
E            +  where 404 = <Response 501 bytes [404 NOT FOUND]>.status_code

tests/test_onionshare_web.py:119: AssertionError
__________________________ TestWeb.test_receive_mode ___________________________

self = <tests.test_onionshare_web.TestWeb object at 0x7f6a18c08358>
common_obj = <onionshare.common.Common object at 0x7f6a18c08438>

    def test_receive_mode(self, common_obj):
        web = web_obj(common_obj, 'receive')
        assert web.mode is 'receive'
    
        with web.app.test_client() as c:
            # Load 404 pages
            res = c.get('/')
            res.get_data()
>           assert res.status_code == 404
E           assert 200 == 404
E            +  where 200 = <Response 1949 bytes [200 OK]>.status_code

tests/test_onionshare_web.py:131: AssertionError
=============================== warnings summary ===============================
/usr/lib/python3.7/site-packages/socks.py:58
  /usr/lib/python3.7/site-packages/socks.py:58: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated, and in 3.8 it will stop working
    from collections import Callable

onionshare_gui/settings_dialog.py:1031
  /tmp/pamac-build/onionshare/src/onionshare-2.0/onionshare_gui/settings_dialog.py:1031: DeprecationWarning: invalid escape sequence \s
    ipv4_pattern = re.compile("(obfs4\s+)?(([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])\.){3}([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5]):([0-9]+)(\s+)([A-Z0-9]+)(.+)$")

onionshare_gui/settings_dialog.py:1032
  /tmp/pamac-build/onionshare/src/onionshare-2.0/onionshare_gui/settings_dialog.py:1032: DeprecationWarning: invalid escape sequence \s
    ipv6_pattern = re.compile("(obfs4\s+)?\[(([0-9a-fA-F]{1,4}:){7,7}[0-9a-fA-F]{1,4}|([0-9a-fA-F]{1,4}:){1,7}:|([0-9a-fA-F]{1,4}:){1,6}:[0-9a-fA-F]{1,4}|([0-9a-fA-F]{1,4}:){1,5}(:[0-9a-fA-F]{1,4}){1,2}|([0-9a-fA-F]{1,4}:){1,4}(:[0-9a-fA-F]{1,4}){1,3}|([0-9a-fA-F]{1,4}:){1,3}(:[0-9a-fA-F]{1,4}){1,4}|([0-9a-fA-F]{1,4}:){1,2}(:[0-9a-fA-F]{1,4}){1,5}|[0-9a-fA-F]{1,4}:((:[0-9a-fA-F]{1,4}){1,6})|:((:[0-9a-fA-F]{1,4}){1,7}|:)|fe80:(:[0-9a-fA-F]{0,4}){0,4}%[0-9a-zA-Z]{1,}|::(ffff(:0{1,4}){0,1}:){0,1}((25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])\.){3,3}(25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])|([0-9a-fA-F]{1,4}:){1,4}:((25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])\.){3,3}(25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9]))\]:[0-9]+\s+[A-Z0-9]+(.+)$")

onionshare_gui/settings_dialog.py:1033
  /tmp/pamac-build/onionshare/src/onionshare-2.0/onionshare_gui/settings_dialog.py:1033: DeprecationWarning: invalid escape sequence \s
    meek_lite_pattern = re.compile("(meek_lite)(\s)+([0-9]+\.[0-9]+\.[0-9]+\.[0-9]+:[0-9]+)(\s)+([0-9A-Z]+)(\s)+url=(.+)(\s)+front=(.+)")

-- Docs: https://docs.pytest.org/en/latest/warnings.html
========= 4 failed, 245 passed, 32 skipped, 4 warnings in 0.88 seconds =========
==> ERROR: A failure occurred in check().
    Aborting...


```

--- a/tests/test_onionshare_web.py
+++ b/tests/test_onionshare_web.py
@@ -41,7 +41,7 @@ RANDOM_STR_REGEX = re.compile(r'^[a-z2-7]+$')
 
 def web_obj(common_obj, mode, num_files=0):
     """ Creates a Web object, in either share mode or receive mode, ready for testing """
-    common_obj.load_settings()
+    common_obj.settings = Settings(common_obj)
     strings.load_strings(common_obj)
     web = Web(common_obj, False, mode)
     web.generate_slug()

